includes/rest-api/class-sensei-rest-api-course-structure-controller.php

```diff
@@ -189,15 +189,15 @@ public function save_course_structure( WP_REST_Request $request ) {
 		}
 
 		$response = new WP_REST_Response();
-		$response->set_data( $course_structure->get( 'edit', wp_using_ext_object_cache() ) );
+		$response->set_data( $course_structure->get( 'edit', wp_using_ext_object_cache(), true ) );
```

Comment:
The third param won't be needed if you remove/add the filter before calling this method. 

```php
remove_filter( 'get_terms', array( Sensei()->modules, 'append_teacher_name_to_module' ), 70 );
$response->set_data( $course_structure->get( 'edit', wp_using_ext_object_cache() ) );
add_filter( 'get_terms', array( Sensei()->modules, 'append_teacher_name_to_module' ), 70, 3 );
```

Reply:
On a side note, I'm not sure why this is done in the `save_course_structure` method. If the remove/add filter is moved to the `get_course_structure` method it still works. Or maybe it should be done in both places? Thoughts?

Reply:
The editor updates the structure based on the json response it gets from the save_course_structure after saving, so if the teacher name property is not there in the response, it removes the teacher name from the UI once we save

Reply:
Thanks! Feels good not having to use another variable to handle only one case ☺️ I've updated it [here](https://github.com/Automattic/sensei/pull/5114/commits/6fa28422a7ab90175748d176224a82a47fec7dc0). You're right about not needing to remove the filter in the `save_course_structure` method. The `is_admin` method returns false here, so the teacher name hook does not append the teacher name to the title.